draft-ietf-jsonpath-base.md

```diff
@@ -510,15 +514,14 @@ Segments can be added to a query to drill further into the structure of the inpu
 The nodelist resulting from the root identifier contains a single node,
 the argument.
 The nodelist resulting from the last segment is presented as the
-result of the query; depending on the specific API, it might be
+result of the query. Depending on the specific API, it might be
 presented as an array of the JSON values at the nodes, an array of
 Normalized Paths referencing the nodes, or both — or some other
 representation as desired by the implementation.
-Note that the API must be capable of presenting an empty nodelist as
-the result of the query.
+Note that an empty nodelist is a valid query result.
 
 A segment performs its function on each of the nodes in its input
-nodelist, during such a function execution, such a node is referred to
+nodelist, during such a function execution, each such a node is referred to
```

Comment:
```suggestion
nodelist. During such a function execution, each such node is referred to
```

However, I think the function and function execution language is clunky, especially when we consider filters containing segments. The scope of the "current node" is lexical rather than dynamic.

Reply:
No biggie, hmm, the definition of "current node" is definitely dynamic as you work your way through the execution, but for any node that is current, its scope is lexical.

Reply:
Actually I've adopted the above text in 95db29d2dde07f3d1fb59435d2754105cca3f62a. We can improve the other language later.